firstvoices/backend/views/search/custom_search_views.py

```diff
@@ -1,12 +1,49 @@
+from drf_spectacular.utils import (
+    OpenApiExample,
+    OpenApiParameter,
+    OpenApiResponse,
+    extend_schema,
+    extend_schema_view,
+    inline_serializer,
+)
 from elasticsearch.exceptions import ConnectionError
-from rest_framework import mixins, viewsets
+from rest_framework import mixins, serializers, viewsets
 from rest_framework.response import Response
 
 from backend.search.query_builder import get_search_query
 from backend.search.utils.object_utils import hydrate_objects
 from backend.views.exceptions import ElasticSearchConnectionError
 
 
+@extend_schema_view(
+    list=extend_schema(
+        description="List of search results satisfying the query.",
+        responses={
+            200: inline_serializer(
+                name="SearchResults",
+                fields={
+                    "id": serializers.CharField(),
+                    "score": serializers.FloatField(),
+                    "type": serializers.CharField(),
+                    "entry": serializers.DictField(),
+                },
+            ),
+            403: OpenApiResponse(description="Todo: Not authorized"),
+        },
+        parameters=[
+            OpenApiParameter(
+                name="q",
+                description="search term",
+                required=False,
+                type=str,
+                examples=[
+                    OpenApiExample("ball", value="ball"),
+                    OpenApiExample("quick brown fox", value="quick brown fox"),
+                ],
+            )
+        ],
+    ),
+)
 class CustomSearchViewSet(mixins.ListModelMixin, viewsets.GenericViewSet):
```

Comment:
I am realizing we called this "custom search" because that's what it is in the old server, but we could probably start fresh and just call it "search"